projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-interface/wpcom-admin-interface.php

```diff
@@ -392,6 +392,37 @@ function wpcom_show_admin_interface_notice() {
 }
 add_action( 'admin_notices', 'wpcom_show_admin_interface_notice' );
 
+/**
+ * Force a cache purge.
+ *
+ * @return void
+ */
+function wpcom_rdv_reset_cache_if_needed() {
+	if ( ! get_user_option( 'rdv_force_cache_is_deleted', get_current_user_id() ) ) {
+		update_user_option( get_current_user_id(), 'rdv_force_cache_is_deleted', true, true );
```

Comment:
Wouldn't this code just delete the cache once? Once we update the `rdv_force_cache_is_deleted` to true, there is nothing that sets it to false. So we will never delete the `RDV_EXPERIMENT_FORCE_ASSIGN_OPTION` after the first time we delete it right?

I'm also failing to understand why we need the new user option (rdv_force_cache_is_deleted), couldn't we just call
`delete_user_option( get_current_user_id(), RDV_EXPERIMENT_FORCE_ASSIGN_OPTION, true );`
on line 449 instead?

Reply:
> So we will never delete the RDV_EXPERIMENT_FORCE_ASSIGN_OPTION after the first time we delete it right

yes, we need to purge the cache just once. After that, the option and the calypso preference will be in sync and it won't be needed.

> I'm also failing to understand why we need the new user option (rdv_force_cache_is_deleted), couldn't we just call
delete_user_option( get_current_user_id(), RDV_EXPERIMENT_FORCE_ASSIGN_OPTION, true );
on line 449 instead?

That would run on every request, basically disabling the cache entirely for a12s. Not only that, it will make an additional call on AT sites for preferences.

Reply:
yes! :)

Reply:
I noticed that **once** the cache is purged and then **re**assigned to `control` via ExPlat (not through the escape hatch), the cache is no longer purged, leading to an inconsistency where Calypso remains in `control` while Jetpack is in `treatment`. 

I don't believe this is a significant issue, as it only occurs in an edge case affecting a12s.

Reply:
That sounds like a bug in the Calypso code 🤔. In theory, the Calypso preference should replace the ExPlat option if it's set.

Reply:
Ah, sorry, you're right! I manually cleared the preference via the Console for some reason during testing, which caused the inconsistency. In a real scenario, the cache and preference should stay aligned.